### ISSUE TITLE
S101 should not report on partial class names where one part is generated

### ISSUE THREAD

**Issue by tamasvajk** [SonarSource/sonar-csharp#908]
S101 reports on the following:

file1.cs:
```C#
public partial class ELN // class name reported
{
}
```

file2.cs
```C#
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------
public partial class ELN
{
...
}
```

**Comment by Evangelink** [comment]
Duplicate #318
